packages/vc-verification/src/issuer-verification.ts

```diff
@@ -0,0 +1,216 @@
+import * as jwt from 'jsonwebtoken';
+import { providers } from 'ethers';
+import { ProofVerifier } from '@ew-did-registry/claims';
+import { Resolver, addressOf } from '@ew-did-registry/did-ethr-resolver';
+import { RegistrySettings } from '@ew-did-registry/did-resolver-interface';
+import { CredentialResolver, IssuerDefinitionResolver } from '.';
+import {
+  IVerifiableCredential,
+  VerificationResult,
+  OffChainClaim,
+} from './models';
+
+export class IssuerVerification {
+  private _resolver: Resolver;
+  private _issuerDefResolver: IssuerDefinitionResolver;
+  private _credentialResolver: CredentialResolver;
+
+  /**
+   *
+   * @param provider
+   * @param registrySetting
+   * @param credentialResolver
+   * @param issuerDefResolver
+   */
+  constructor(
+    provider: providers.Provider,
+    registrySetting: RegistrySettings,
+    credentialResolver: CredentialResolver,
+    issuerDefResolver: IssuerDefinitionResolver
+  ) {
+    this._issuerDefResolver = issuerDefResolver;
+    this._resolver = new Resolver(provider, registrySetting);
+    this._credentialResolver = credentialResolver;
+  }
+
+  /**
+   * Verifies chain of trust for a holder's credential
+   * @param credential
+   * @param verifyCredentialProofCallback verification callback approach to verify all issuers in hierarchy. By default the chain is verified against RoleDefinition
+   * @returns
+   */
+  async verifyChainOfTrust(
+    credential: IVerifiableCredential,
+    verifyCredentialProofCallback?: (
+      credential: IVerifiableCredential
+    ) => Promise<VerificationResult>
+  ) {
+    if (credential && verifyCredentialProofCallback) {
+      return this.verifyChainOfTrustCallback(
+        credential,
+        verifyCredentialProofCallback
+      );
+    } else if (credential) {
+      return this.verifyChainOfTrustByRoleDefinition(credential);
+    } else {
+      throw new Error('Not sufficient information');
+    }
+  }
+
+  /**
+   * Verifies chain of trust for a given holder's DID and role
+   * @param {string} credential
+   */
+  async verifyChainOfTrustByRoleDefinition(credential: IVerifiableCredential) {
+    let isVcVerified = false;
+    let hasParent = true;
```

Comment:
Last small suggestion. The return semantic of this method is not consistent. On failed verification method throws, on successful method returns boolean. I think on success we can just return from method